driver/terraform.go

```diff
@@ -96,7 +97,17 @@ func NewTerraform(config *TerraformConfig) (*Terraform, error) {
 		log.Printf("[ERR] (driver.terraform) init client type '%s' error: %s", config.ClientType, err)
 		return nil, err
 	}
-	if env := config.Task.Providers.Env(); len(env) > 0 {
+
+	if taskEnv := config.Task.Env; len(taskEnv) > 0 {
+		// Terraform init requires discovering git in the PATH env.
+		//
+		// The terraform-exec package disables inheriting from the os environment
+		// when using tfexec.SetEnv(). So for CTS purposes, we'll force inheritance
+		// to allow Terraform commands to use the os environment as necessary.
+		env := envMap(os.Environ())
+		for k, v := range taskEnv {
+			env[k] = v
+		}
```

Comment:
Helpful comment! Also personally like inheriting os env vars close to the tfClient.SetEnv

Reply:
That's good to hear, thanks.

I also found that if this was done in the controller, it would have made the tests a bit more tedious to maintain. https://app.circleci.com/pipelines/github/hashicorp/consul-terraform-sync/711/workflows/f577eac6-6266-432d-8559-ff85aa75fa1c/jobs/1380/parallel-runs/0/steps/0-104?invite=true

Reply:
Good call on keeping code oriented toward being testable. :100: